### ISSUE TITLE
problems with v1.0.0-RC10

### ISSUE THREAD

**Issue by eventhorizonpl** [sphereio/commercetools-php-sdk#187]
Hi,

While trying to update to v1.0.0-RC10 I noticed a problem.

I think it might be related to redis, but I don't know how exactly.

$product object is cached in redis for 60 seconds

```
            $product_data = json_decode(strip_tags(LocalizerService::getLocalized($product->getMasterData()->getCurrent()->getDescription(), $request->getLocale())), true);

            if (null === $product_data) {
                throw new NotFoundHttpException('no description for product id ('.$product_id.')');
            }
```

If I visit the product page sometimes I get this error, sometimes no.

This behaviour doesn't occur in  v1.0.0-RC9.

Any ideas?

Best regards,
Michal

**Comment by jayS-de** [comment]
I would need more detailed information about your problem. We could start a direct chat using our public support channel: https://www.hipchat.com/gCOStFSHE

**Comment by jayS-de** [comment]
Okay, after some analysis we found the problem.

In [RC10](https://github.com/sphereio/commercetools-php-sdk/commit/621195d) I fixed a bug for direct access to a language in the LocalizedString object. Now the LocalizedString correctly uses the Graceful-Flag. This broke the workaround code in the getLocalized function of the LocalizerService written by @eventhorizonpl .

To fix the issue he now uses something like this:
```
function getLocalized($object, $locale)
{
    $fallbackLocale = 'fr';
    $context = Context::of()->setLanguages([$locale, $fallbackLocale])->setGraceful(true);

    return (string)$object->get($context);
}
```
